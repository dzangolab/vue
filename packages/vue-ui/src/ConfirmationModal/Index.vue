<template>
  <div :class="modalClassName">
    <div class="modal-wrapper" @click="handleClose()">
      <div class="modal-container">
        <div class="modal-header">
          <slot v-if="!disableHeader" name="header">
            <span>Confirmation required</span>
          </slot>

          <svg
            xmlns="http://www.w3.org/2000/svg"
            width="1em"
            height="1em"
            viewBox="0 0 16 16"
            @click="handleClose()"
          >
            <path
              fill="currentColor"
              d="M5.28 4.22a.75.75 0 0 0-1.06 1.06L6.94 8l-2.72 2.72a.75.75 0 1 0 1.06 1.06L8 9.06l2.72 2.72a.75.75 0 1 0 1.06-1.06L9.06 8l2.72-2.72a.75.75 0 0 0-1.06-1.06L8 6.94z"
            />
          </svg>
        </div>

        <div v-if="!disableBody" class="modal-body">
          <slot name="body">
            <p>Are you sure you want to proceed?</p>
          </slot>
        </div>

        <div v-if="!disableFooter" class="modal-footer">
          <slot name="footer">
            <ButtonElement
              size="small"
              label="Cancel"
              severity="secondary"
              @click="handleClose()"
            />

            <ButtonElement
              size="small"
              label="Confirm"
              severity="primary"
              @click="handleConfirm()"
            />
          </slot>
        </div>
      </div>
    </div>
  </div>
</template>

<script lang="ts">
export default {
  name: "ConfirmationModal",
};
</script>

<script setup lang="ts">
import { computed } from "vue";

import ButtonElement from "../Button/Index.vue";

const emits = defineEmits(["on:confirm", "on:close"]);

const props = defineProps({
  border: {
    default: false,
    type: Boolean,
  },
  disableBody: {
    default: false,
    type: Boolean,
  },
  disableHeader: {
    default: false,
    type: Boolean,
  },
  disableFooter: {
    default: false,
    type: Boolean,
  },
});

const modalClassName = computed(() => {
  return ["modal", props.border && "bordered-modal"].filter(Boolean).join(" ");
});

function handleConfirm() {
  emits("on:confirm");
}

function handleClose() {
  emits("on:close");
}
</script>
