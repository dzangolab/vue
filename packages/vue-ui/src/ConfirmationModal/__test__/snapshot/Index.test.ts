import { mount } from "@vue/test-utils";
import { describe, expect, it } from "vitest";

import ConfirmationModal from "../../Index.vue";

describe("ConfirmationModal", () => {
  it("matches snapshot", () => {
    const wrapper = mount(ConfirmationModal, {
      props: {
        border: true,
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });
});
