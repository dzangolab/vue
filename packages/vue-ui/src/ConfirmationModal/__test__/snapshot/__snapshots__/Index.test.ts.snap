// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ConfirmationModal > matches snapshot 1`] = `
<div
  class="modal bordered-modal"
>
  <div
    class="modal-wrapper"
  >
    <div
      class="modal-container"
    >
      <div
        class="modal-header"
      >
        
        <span>
          Confirmation required
        </span>
        
        <svg
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M5.28 4.22a.75.75 0 0 0-1.06 1.06L6.94 8l-2.72 2.72a.75.75 0 1 0 1.06 1.06L8 9.06l2.72 2.72a.75.75 0 1 0 1.06-1.06L9.06 8l2.72-2.72a.75.75 0 0 0-1.06-1.06L8 6.94z"
            fill="currentColor"
          />
        </svg>
      </div>
      <div
        class="modal-body"
      >
        
        <p>
          Are you sure you want to proceed?
        </p>
        
      </div>
      <div
        class="modal-footer"
      >
        
        <button
          class="button secondary small filled"
          role="button"
        >
          <!--v-if-->
          <div
            class="label"
          >
            
            <span>
              Cancel
            </span>
            
          </div>
          <!--v-if-->
          <!--v-if-->
        </button>
        <button
          class="button primary small filled"
          role="button"
        >
          <!--v-if-->
          <div
            class="label"
          >
            
            <span>
              Confirm
            </span>
            
          </div>
          <!--v-if-->
          <!--v-if-->
        </button>
        
      </div>
    </div>
  </div>
</div>
`;
